types/sshpk/index.d.ts

```diff
@@ -103,7 +103,22 @@ declare namespace SshPK {
         ca?: boolean;
     }
 
-    type CertificateFormat = 'openssh' | 'pem' | 'x509';
+    type CertificateFormats = 'openssh' | 'pem' | 'x509';
+
+    interface CertificateFormat extends Format {
+        verify(cert: Certificate, key: Key): boolean;
+        sign(cert: Certificate, key: Key): boolean;
+    }
+
+    interface x509Format extends CertificateFormat {
+        signAsync(cert: Certificate, signer: SignerFunction, done: (err: Error | undefined) => void): void;
+    }
+
+    interface OpenSSHCertificateFormat extends CertificateFormat {
+        signAsync(cert: Certificate, signer: SignerFunction, done: (err: Error | undefined) => void): void;
+        fromBuffer(data: Buffer, algo?: AlgorithmType, partial?: SSHBuffer): Certificate;
```

Comment:
Functions `fromBuffer` and `toBuffer` are marked as internal private API. I think that exposing private API is not a good idea. 

Besides, types for `algo` and `partial` arguments are wrong. Supported algorithms in OpenSSH certificates are not the same as defined in `AlgorithmType` (see https://cvsweb.openbsd.org/src/usr.bin/ssh/PROTOCOL.certkeys?annotate=HEAD and https://github.com/TritonDataCenter/node-sshpk/blob/3e8366433e9c2ef688e7b14935bd4d4fa264229d/lib/formats/openssh-cert.js#LL330C28-L330C28). `partial` argument is not `SSHBuffer`, but probably `{ remainder?: Buffer, consumed?: number }`, because this argument is not used in any call of this function, but same pattern is used in `formats/ssh-private.js` (see https://github.com/TritonDataCenter/node-sshpk/blob/3e8366433e9c2ef688e7b14935bd4d4fa264229d/lib/formats/ssh-private.js#LL126C7-L126C7).